.circleci/tag.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -e
```

Comment:
It looks like this script has a lot of overlap with `.circleci/deploy.sh`. Is it possible to eliminate the duplication, or perhaps even to eliminate one of the scripts with params or whatever?

Reply:
There is, but I didn't see a clear way to pull apart the steps, so I made a new script that merged the desired steps and checked for a new Dockerfile-tag that needs to function differently.